### PR TITLE
Spanish translations

### LINKED ISSUE #135: Spanish translation tracking
# English

## Description

- This issue is for tracking the translation progress of all pages.
- This issue will be closed when the documentation is finished.

## I want to collaborate

1. Being a translator

- Read [how to contribute in translations](https://github.com/grammyjs/website/blob/main/CONTRIBUTING.md#translations)
- Translate with your native-speaker level, **DO NOT** just copy the section to some translator tool and paste back the result
- Take one untranslated page, translate and create PR from your fork to the main project. One PR for page, to better split in parallel the work and less tired reviews

2. Being a reviewer

- Find the PR's created for translations and review them. Add all notes that you consider.

## Note

Words related to methods, function names, variables... mustn't be translated. Here are some examples:
- bot
- update
- keyboard / inline keyboard
- webhook
- payload
- ... (feel free to add more, but make sense)

---

# Español

## Descripción

- Esta issue es para rastrear el progreso de traducción de todas las páginas.
- Está issue será cerrada cuando la documentación este completada.

## Me gustaría colaborar

1. Como traductor

- Leéte [como contribuir en las traducciones](https://github.com/grammyjs/website/blob/main/CONTRIBUTING.md#translations)
- Traduce con tu nivel nativo, **NO** simplemente copies el contenido en una herramienta de traducción y pegues el resultado
- Escoge una página no traducida, traducela y crea un PR desde el fork del projecto al projecto principal. Un PR por página, para favorecer la parallelización del trabajo y reducir el cansancio de la revisión.

2. Como validador

- Encuentra un PR creado de traducciones y revisalo. Añade todos los comentarios que consideres.

## Nota

Palabras relativas a métodos, nombres de funciones, variables... no se traducen. Aquí algunos ejemplos:

- bot
- update
- keyboard / inline keyboard
- webhook
- payload
- ... (añade más si consideras, pero con sentido común)

---

## Readme

- [README.md](https://github.com/grammyjs/website/blob/main/site/docs/es/README.md) -> #319

## Advanced

- [deployment.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/deployment.md) -> #293 
- [flood.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/flood.md) -> #294 
- [middleware.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/middleware.md) -> #295 
- [proxy.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/proxy.md) -> #296
- [README.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/README.md) -> #297
- [reliability.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/reliability.md) -> #298
- [scaling.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/scaling.md) -> #299
- [structuring.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/structuring.md) -> #300
- [transformers.md](https://github.com/grammyjs/website/blob/main/site/docs/es/advanced/transformers.md) -> #301

## Demo

- [examples.md](https://github.com/grammyjs/website/blob/main/site/docs/es/demo/examples.md) -> #302
- [README.md](https://github.com/grammyjs/website/blob/main/site/docs/es/demo/README.md) -> #310

## Guide

- [api.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/api.md) -> #303
- [basics.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/basics.md) -> #304
- [commands.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/commands.md) -> #305 
- [context.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/context.md) -> #306
- [deployment-types.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/deployment-types.md) -> #307
- [errors.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/errors.md) -> #308
- [files.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/files.md) -> #309
- [filter-queries.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/filter-queries.md) ->  #311
- [games.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/games.md) -> #312
- [getting-started.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/getting-started.md) -> #313
- [inline-queries.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/inline-queries.md) -> #314
- [introduction.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/introduction.md) -> #211 
- [middleware.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/middleware.md)
    -> #212
    -> #335
- [README.md](https://github.com/grammyjs/website/blob/main/site/docs/es/guide/README.md) -> #318 

## Hosting

- [deno-deploy.md](https://github.com/grammyjs/website/blob/main/site/docs/es/hosting/deno-deploy.md)
    -> #231 
    -> #357
- [gcf.md](https://github.com/grammyjs/website/blob/main/site/docs/es/hosting/gcf.md) -> #232 
- [heroku.md](https://github.com/grammyjs/website/blob/main/site/docs/es/hosting/heroku.md) -> #235 
- [vps.md](https://github.com/grammyjs/website/blob/main/site/docs/es/hosting/vps.md) -> #344

## Plugins

- [auto-retry.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/auto-retry.md) -> #236 
- [command-filter.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/command-filter.md) -> #237 
- [console-time.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/console-time.md) -> #238 
- [files.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/files.md) -> #239 
- [hydratate.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/hydratate.md) -> #240 
- [i18n.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/i18n.md) -> #244 
- [keyboard.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/keyboard.md) -> #253
- [menu.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/menu.md) -> #254 
- [parse-mode.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/parse-mode.md) -> #255
- [ratelimiter.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/ratelimiter.md)
    -> #256
    -> #356
- [README.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/README.md) -> #257
- [router.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/router.md) -> #258
- [runner.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/runner.md) -> #259
- [session.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/session.md) -> #260
- [stateless-question.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/stateless-question.md) -> #261
- [transformer-throttler.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/transformer-throttler.md) -> #262
- [emoji.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/emoji.md) -> #345
- [fluent.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/fluent.md) -> #346
- [guide.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/guide.md) -> #347
- [middlewares.md](https://github.com/grammyjs/website/blob/main/site/docs/es/plugins/middlewares.md) -> #349

## Resources

- [comparison.md](https://github.com/grammyjs/website/blob/main/site/docs/es/resources/comparison.md) -> #263
- [faq.md](https://github.com/grammyjs/website/blob/main/site/docs/es/resources/faq.md) -> #264
- [about.md](https://github.com/grammyjs/website/blob/main/site/docs/es/resources/about.md) -> #343

### PR DESCRIPTION
**Pull request by KnorpelSenf**
- Closes #135.
- [x] Fix: The links to next or previous chapters at the bottom of the pages sometimes switch back to English. Please make sure that all links keep the user in the Spanish section of the docs. One example of where this reproduces is to navigate back to the overview of the Guide in the Learn section. I am aware of more places, so please check all pages.
- [x] Fix: The image for filter query search is missing.
- [x] Fix: The first two links in the FAQ page are broken.
- [x] Fix: The link Resources > About grammY is broken in the Spanish menu.
- [x] Reassure: IIRC, there are more broken prev/next links in the other parts of the docs, not just in the Guide. Did you check?
- [x] Outdated: `es/plugins/README.md` is outdated. See https://github.com/grammyjs/website/commit/d135978733b343b3f9929dda8f1c03044226b13c.
- [x] Translate: The title “Long Polling vs. Webhooks” isn't translated. I understand that those are two technical terms and it's fine if they are kept in English—I'd do the same for German—but the “versus” probably has a better translation. (Correct me if I'm wrong.)
- [x] Translate: Middleware Redux means something like “Middleware Revisited” or “Middleware Once Again” so it should also have a proper translation. Redux isn't a technical term here.
- [x] Translate: The 404 page, configured in the config.ts file, contains jokes that don't translate well. For example, the haiku error message is actually a haiku in English, with the correct number of syllables for each line. However, this joke breaks in Spanish, as the direct translation isn't a haiku anymore. Similarly, for Chinese, the song Country Roads, Take Me Home isn't familiar to the audience, so it had to be changed to a poem (IIRC) that's known in the Chinese-speaking community and still makes sense in the context. Spanish has the same freedom here: it doesn't have to be a haiku, and it doesn't even have to be the same number of error messages. Just write your own list, come up with something you find amusing or that you think will make people think or smile or both. It's an Easter Egg here, so inside jokes are explicitly allowed.

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
@habemuscode @roj1512 we're missing Spanish in the settings, so the vuepress config isn't updated yet. Perhaps you're already aware?

**Comment by roj1512** [comment]
I'm not. @habemuscode please work on it.

**Comment by roj1512** [comment]
@KnorpelSenf Do you know why the index page of the [preview](https://sour-newt-93-8mphj3ee9gdg.deno.dev/) is in Spanish, and not English?

**Comment by KnorpelSenf** [comment]
> @KnorpelSenf Do you know why the index page of the [preview](https://6279615cc1aff2173b38deae--grammy.netlify.app/) is in Spanish, and not English?

No. This does not make sense. The deployment does not correspond to the docs.

Can you reproduce it locally?

This hasn't happened before we installed the sitemap CLI. Perhaps it's related?

**Comment by roj1512** [comment]
> This hasn't happened before ...

Actually, the index page was recently translated into Spanish. That's a thing, too.

> Can you reproduce it locally?

Yes, it is the same locally.

--

Also, if I rename the `es` directory to something else, like `zd`, the index page will be back in English.

**Comment by KnorpelSenf** [comment]
Sounds like a VuePress bug then?

**Comment by roj1512** [comment]
Requires #350 and #338.

**Comment by KnorpelSenf** [comment]
#338 and #350 are already included, correct? This means that they should receive Spanish flag tags.

**Comment by roj1512** [comment]
I didn't know we were already in that state.

**Comment by KnorpelSenf** [comment]
> I didn't know we were already in that state.

You mean you weren't aware we're done? 😁

**Comment by roj1512** [comment]
I mean I thought those labels work only when the modification PR itself is actually updated to support Spanish.

**Comment by KnorpelSenf** [comment]
Ohh I see that point. Yeah true, you're right, we should start doing stuff like that once we enter the maintenance stage.

**Review by KnorpelSenf — COMMENTED** [review]
Here are a few things that are broken.

1. The links to next or previous chapters at the bottom of the pages sometimes switch back to English. Please make sure that all links keep the user in the Spanish section of the docs. One example of where this reproduces is to navigate back to the overview of the Guide in the Learn section. I am aware of more places, so please check all pages.
2. The image for filter query search is missing.
3. The title “Long Polling vs. Webhooks” isn't translated. I understand that those are two technical terms and it's fine if they are kept in English—I'd do the same for German—but the “versus” probably has a better translation. (Correct me if I'm wrong.)
4. The 404 page, configured in the config.ts file, contains jokes that don't translate well. For example, the haiku error message is actually a haiku in English, with the correct number of syllables for each line. However, this joke breaks in Spanish, as the direct translation isn't a haiku anymore. Similarly, for Chinese, the song Country Roads, Take Me Home isn't familiar to the audience, so it had to be changed to a poem (IIRC) that's known in the Chinese-speaking community and still makes sense in the context. Spanish has the same freedom here: it doesn't have to be a haiku, and it doesn't even have to be the same number of error messages. Just write your own list, come up with something you find amusing or that you think will make people think or smile or both. It's an Easter Egg here, so inside jokes are explicitly allowed.

With these things out of the way, I will have to repeat myself and express that this is truly great work. You can be proud of this, what an accomplishment! 🎉

**Comment by roj1512** [comment]
@habemuscode I've eliminated problems 1 and 2.

Now, you can focus on the 3rd and 4th problems:
 
>     3. The title “Long Polling vs. Webhooks” isn't translated. I understand that those are two technical terms and it's fine if they are kept in English—I'd do the same for German—but the “versus” probably has a better translation. (Correct me if I'm wrong.)
> 
>     4. The 404 page, configured in the config.ts file, contains jokes that don't translate well. For example, the haiku error message is actually a haiku in English, with the correct number of syllables for each line. However, this joke breaks in Spanish, as the direct translation isn't a haiku anymore. Similarly, for Chinese, the song Country Roads, Take Me Home isn't familiar to the audience, so it had to be changed to a poem (IIRC) that's known in the Chinese-speaking community and still makes sense in the context. Spanish has the same freedom here: it doesn't have to be a haiku, and it doesn't even have to be the same number of error messages. Just write your own list, come up with something you find amusing or that you think will make people think or smile or both. It's an Easter Egg here, so inside jokes are explicitly allowed.
> 
>

**Comment by KnorpelSenf** [comment]
What happened during the force-push?

IIRC, there are more broken prev/next links in the other parts of the docs, not just in the Guide. Did you check?

**Comment by KnorpelSenf** [comment]
I think the other links were also fixed?

I discovered one more thing:

5. Middleware Redux means something like “Middleware Revisited” or “Middleware Once Again” so it should also have a proper translation. Redux isn't a technical term here.

**Comment by KnorpelSenf** [comment]
6. The link Resources > About grammY is broken in the Spanish menu.

**Comment by KnorpelSenf** [comment]
7. The first two links in the FAQ page are broken.

**Comment by roj1512** [comment]
> What happened during the force-push?

I changed my commit message, there was a typo in it.

**Comment by roj1512** [comment]
- [x] The links to next or previous chapters at the bottom of the pages sometimes switch back to English. Please make sure that all links keep the user in the Spanish section of the docs. One example of where this reproduces is to navigate back to the overview of the Guide in the Learn section. I am aware of more places, so please check all pages.
- [x] The image for filter query search is missing.
- [ ] The title “Long Polling vs. Webhooks” isn't translated. I understand that those are two technical terms and it's fine if they are kept in English—I'd do the same for German—but the “versus” probably has a better translation. (Correct me if I'm wrong.)
- [ ] The first two links in the FAQ page are broken.
- [ ] The link Resources > About grammY is broken in the Spanish menu.
- [ ] Middleware Redux means something like “Middleware Revisited” or “Middleware Once Again” so it should also have a proper translation. Redux isn't a technical term here.
- [ ] IIRC, there are more broken prev/next links in the other parts of the docs, not just in the Guide. Did you check?

**Comment by roj1512** [comment]
@habemuscode Also, I just saw that `es/plugins/README.md` is outdated. See https://github.com/grammyjs/website/commit/d135978733b343b3f9929dda8f1c03044226b13c.

**Comment by KnorpelSenf** [comment]
The first two links in the plugins overview (`/es/plugins/`) redirect to the English pages. Is there maybe a regex that lets us ascertain that all links on Spanish pages stay on Spanish pages?

**Comment by roj1512** [comment]
@KnorpelSenf I’ve fixed all links that were redirecting to English.

**Comment by KnorpelSenf** [comment]
In fact, Chinese has the same problem of broken links that redirect to English. For example, the link from the menu to the API transformers, far down at the bottom: https://grammy.dev/zh/plugins/menu.html#%E5%AE%83%E6%98%AF%E5%A6%82%E4%BD%95%E5%B7%A5%E4%BD%9C%E7%9A%84

Perhaps we need a more general cleanup of these links. However, this isn't blocking Spanish as we will fix it afterwards anyway. Moreover, we do not need any translators to take care of this.

**Comment by roj1512** [comment]
> IIRC, there are more broken prev/next links in the other parts of the docs, not just in the Guide. Did you check?

@KnorpelSenf I went through the other part which has prev/next links, namely the Advanced part, I couldn't find broken ones. Is there anything I might be msising?

**Comment by KnorpelSenf** [comment]
I think it's been fixed in one of your commits already.

**Comment by KnorpelSenf** [comment]
Or at least I can't find it either anymore :D

**Comment by KnorpelSenf** [comment]
@roj1512 are you able to resolve the conflicts?

**Comment by KnorpelSenf** [comment]
I saw you added a to-do list to the description. Should point 4. be added there?

**Comment by roj1512** [comment]
@KnorpelSenf Done.

**Comment by KnorpelSenf** [comment]
Perfect. I think this can be merged as soon as the last three things are translated @habemuscode

File: site/docs/.vuepress/config.ts [hunk 1]
```diff
@@ -369,11 +369,11 @@ export default defineUserConfig({
...
         lastUpdatedText: "Última actualización",
         notFound: [
-          "No se ha encontrado",
+          "¡Uy! Esta página no existe",
           "Nop.",
-          "nada aquí para ti, lo siento",
-          "Error 404/ Esta página no se pudo encontrar pero/ Un haiku en su lugar",
-          "Caminos ruuuuraaaales",
+          "Nada aquí para ti, lo siento",
+          "Error 404/ ¡Houston, tenemos un problema!",
```
Comment (KnorpelSenf) [inline site/docs/.vuepress/config.ts:375]: Hehe nice one

Comment (habemuscode) [reply site/docs/.vuepress/config.ts:375]: :D

**Review by roj1512 — APPROVED** [review]
El thankos per tu efforte!

**Pull request merged**